[BZ1810461] ShiftStack - Add procedure to allow Image Registry Operator to trust Swift (#39641)
openshift/openshift-docs: modules/registry-configuring-registry-storage-swift-trust.adoc

// Module included in the following assemblies:
//
// * registry/installing-openstack- .adoc
// * registry/configuring-registry-operator.adoc
// * registry/configuring-registry-storage-openstack-user-infrastructure.adoc

> 

Author: ```suggestion
:_content-type: PROCEDURE
```

[id="registry-configuring-registry-storage-swift-trust_{context}"]
= Configuring the Image Registry Operator to trust Swift storage

You must configure the Image Registry Operator to trust {rh-openstack-first} Swift storage.

// to allow the client to pull the image layers from the image registry rather than from links directly from Swift.

> 

Reviewer: Does this procedure have prereqs? It looks like there's nothing to change in the command itself, so I suspect there might be. You might want to add them at some point.

.Procedure

> * To configure the Image Registry Operator to trust OpenStack Swift storage, in the `config.imageregistry` object, change the value of the `spec.disableRedirect` field to `true`.

Author: ```suggestion
* From a command line, enter the following command to change the value of the `spec.disableRedirect` field in the  `config.imageregistry` object to `true`:
```

> * From a command line, enter the following command to change the value of the `spec.disableRedirect` field in the  `config.imageregistry` object to `true`:

Reviewer: ```suggestion
* From a command line, enter the following command to change the value of the `spec.disableRedirect` field in the `config.imageregistry` object to `true`:
```

// Module included in the following assemblies:
//
// * registry/installing-openstack- .adoc
// * registry/configuring-registry-operator.adoc
// * registry/configuring-registry-storage-openstack-user-infrastructure.adoc

[id="registry-configuring-registry-storage-swift-trust_{context}"]
= Configuring the Image Registry Operator to trust Swift storage

You must configure the Image Registry Operator to trust {rh-openstack-first} Swift storage.

// to allow the client to pull the image layers from the image registry rather than from links directly from Swift.

.Procedure

* To configure the Image Registry Operator to trust OpenStack Swift storage, in the `config.imageregistry` object, change the value of the `spec.disableRedirect` field to `true`.

> 

Author: ```suggestion
```

+
[source,terminal]
----

> $ oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":"true"}}'

Reviewer 2: The cmd shoud be, true value is a boolean 
oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":true}}'

Author: ```suggestion
$ oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":true}}'
```

@xiuwang All good if this is committed?

Reviewer 2: Yes, other LGTM

// Module included in the following assemblies:
//
// * registry/installing-openstack- .adoc
// * registry/configuring-registry-operator.adoc
// * registry/configuring-registry-storage-openstack-user-infrastructure.adoc

[id="registry-configuring-registry-storage-swift-trust_{context}"]
= Configuring the Image Registry Operator to trust Swift storage

> You must configure the Image Registry Operator to trust OpenStack Swift storage.

Author: ```suggestion
You must configure the Image Registry Operator to trust {rh-openstack-first} Swift storage.
```

////
...to allow the client to pull the image layers from the image registry rather than from links directly from Swift.
////

.Procedure

> * To configure the Image Registry Operator to trust OpenStack Swift storage, change the ____ field on the `config.imageregistry` object to `spec.disableRedirect: true`.

Reviewer 3: It's a draft, it should be a complete sentence.

Author: ```suggestion
* To configure the Image Registry Operator to trust OpenStack Swift storage, in the `config.imageregistry` object, change the value of the  `spec.disableRedirect` field to `true`.
```

+

> [source,terminal]

Reviewer 4: According to https://bugzilla.redhat.com/show_bug.cgi?id=1810461#c13, the following should be all that is needed:

```
$ oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":"true"}}'
```

Author: @mandre As in that command should be almost the entirety of this PR? Easy enough, if so. 

Any thoughts on the second part of this comment? https://bugzilla.redhat.com/show_bug.cgi?id=1810461#c14

Reviewer 4: > @mandre As in that command should be almost the entirety of this PR? Easy enough, if so.
> 
> Any thoughts on the second part of this comment? https://bugzilla.redhat.com/show_bug.cgi?id=1810461#c14

If I understand comment 14 correctly what we're doing here (disable redirect) is what the customer has confirmed fixes the issue. I do not think we want to document the other suggested solution as it's a lot more complex and forces the nodes to trust the CA ultimately (while we trust it only when initiating connections to OpenStack otherwise).

----

> $ oc create configmap registry-config --from-file=<external_registry_address>=ca.crt -n openshift-config

Reviewer 3: Is it a necessary step?

> $ oc create configmap registry-config --from-file=<external_registry_address>=ca.crt -n openshift-config

Author: ```suggestion
$ oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":"true"}}'
```

----

[source,terminal]
----

> oc edit image.config.openshift.io cluster

Author: ```suggestion
$ oc edit image.config.openshift.io cluster
```

----

> [source,yaml]
> spec:

Author: ```suggestion
[source,yaml]
----
spec:
```

  additionalTrustedCA:
    name: registry-config
spec

>   disableRedirect: true

Reviewer 3: disableRedirect is in `config.imageregistry`, not `image.config`.

// Module included in the following assemblies:
//
// * registry/installing-openstack- .adoc
// * registry/configuring-registry-operator.adoc
// * registry/configuring-registry-storage-openstack-user-infrastructure.adoc

[id="registry-configuring-registry-storage-swift-trust_{context}"]
= Configuring the Image Registry Operator to trust Swift storage

You must configure the Image Registry Operator to trust {rh-openstack-first} Swift storage.

////
...to allow the client to pull the image layers from the image registry rather than from links directly from Swift.
////

.Procedure

* To configure the Image Registry Operator to trust OpenStack Swift storage, in the `config.imageregistry` object, change the value of the  `spec.disableRedirect` field to `true`.

+
[source,terminal]
----
$ oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"disableRedirect":"true"}}'
----

[source,terminal]
----

> $ oc edit image.config.openshift.io cluster

Author: @mandre Could you confirm this command given this comment? https://github.com/openshift/openshift-docs/pull/39641/files#r765059344

Author: TY!

Author: @mandre So, just this? https://github.com/openshift/openshift-docs/pull/39641/commits/3c855cef84b70b974ecc6907d197e47f5dd12430

Reviewer 4: Yup, perfect :+1:
